### PR TITLE
zson: Pretty print type values

### PR DESCRIPTION
**Pull request by mattnibs**
Add indenting and coloring when formatting type values. Also
standardizes all formatting of type values into single implementation.

### CODE REVIEW

**Comment by mccanne** [comment]
A couple thoughts
* The color scheme is a little jarring to me... how about we live it as gray then we can experiment with colored approaches and we use it more
* Array, set, and map types will typically be simple, e.g.,  I don't think `[int64]` should be vertically formatted.  Maybe we could have a heuristic that vertically formats these only when there is a compound type inside, e.g., array of object or even array of array could be 
```
[
    [int64]
]
```

**Comment by mattnibs** [comment]
@mccanne updated, I took away the colors and gave primitives a slightly darker gray:
image

**Comment by mattnibs** [comment]
Also I just based the heuristic for indenting arrays, sets, maps based off of if the inner type is a primitive. I could enable this also for enums, errors, named types but this didn't seem quite as straightforward, not to mention unraveling to types for maps where you need to find the type for the key and value is a harder problem.

File: zson/formatter.go [hunk 1]
```diff
@@ -206,6 +208,174 @@ func (f *Formatter) formatValue(indent int, typ zed.Type, bytes zcode.Bytes, par
...
+			return nil
+		}
+		for k := 0; k < n; k++ {
+			if k > 0 {
+				f.build(",")
+			}
+			var symbol string
+			symbol, tv = zed.DecodeName(tv)
+			if tv == nil {
+				return nil
```
Comment (nwt) [inline zson/formatter.go:339]:
```suggestion
				f.truncTypeValueErr()
				return nil
```

File: zson/formatter.go [hunk 2]
```diff
@@ -206,6 +208,174 @@ func (f *Formatter) formatValue(indent int, typ zed.Type, bytes zcode.Bytes, par
...
+	return tv
+}
+
+func (f *Formatter) formatVectorTypeValue(token string, indent int, tv zcode.Bytes) zcode.Bytes {
+	f.build(token)
+	end := "]"
+	if token == "|[" {
+		end += "|"
+	}
+	if n, _ := zed.DecodeLength(tv); n <= zed.IDTypeComplex {
```
Comment (nwt) [inline zson/formatter.go:361]: Check for an error and call `Formatter.truncTypeValueErr`.

File: zson/formatter.go [hunk 3]
```diff
@@ -206,6 +208,174 @@ func (f *Formatter) formatValue(indent int, typ zed.Type, bytes zcode.Bytes, par
...
+		tv = f.formatVectorTypeValue("[", indent, tv)
+	case zed.TypeValueSet:
+		tv = f.formatVectorTypeValue("|[", indent, tv)
+	case zed.TypeValueMap:
+		f.build("|{")
+		newline := f.newline
+		indent += f.tab
+		if n, itv := zed.DecodeLength(tv); n <= zed.IDNull {
+			n, _ = zed.DecodeLength(itv)
+			if n <= zed.IDNull {
```
[the remark this answers is not in the document]
Comment (mattnibs) [reply zson/formatter.go:286]: The errors from decode length will get caught below when this is actually rendered.

[the remark this answers is not in the document]
Comment (mattnibs) [reply zson/formatter.go:286]: Also curious why `< zed.IDTypeComplex` is superior to `<= zed.IDNull`?

File: zson/formatter.go (hunk 2 above)
Comment (mattnibs) [reply zson/formatter.go:361]: No need, as this will get caught on the call to `formatTypeValue`.

Comment (mattnibs) [reply zson/formatter.go:362]: But this should be `n < zed.IDTypeComplex`

File: zson/formatter.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (nwt) [reply zson/formatter.go]: Unlike `IDNull`, it's robust to addition of primitive types. If we ever add more, `IDTypeComplex` will change accordingly.

File: zson/ztests/pretty-type-value.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
+zed: "yield this, '==='"
```
Comment (nwt) [inline zson/ztests/pretty-type-value.yaml:1]:
```suggestion
zed: yield this, "==="
```

File: zson/ztests/pretty-type-value.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+input: |
+  <({foo:int64},{bar:string},uint64)>
+  <named=|[({foo:int64},uint64)]|>
+  <|{{x:string}:{ip:ip,port:uint16}}|>
+  <|{string:uint64}|>
+  <[int64]>
+  <error([string])>
+  <[[string]]>
+
+output-flags: "-pretty=4"
```
Comment (nwt) [inline zson/ztests/pretty-type-value.yaml:12]:
```suggestion
output-flags: -pretty=4
```

**Review by nwt — APPROVED** [review]

**Pull request merged**